### PR TITLE
Makefile.am, configure.ac: fix hard coded static and include path

### PR DESCRIPTION
**Pull request by vjardin**
This serie is used to avoid hard coded path that do not work for any distributions. For instance, when buildroot is used or when cross compilation is used, the path of the headers cannot be enforce with /usr/include .

### CODE REVIEW

**Comment by vjardin** [comment]
Side note : without this patch, it cannot link because 
libmtcr_ul.a is not available, only libmtcr_ul.so (or its libmtcr_ul.la) had been compiled.

**Comment by vjardin** [comment]
Meanwhile, this pull request is included with the Buildroot package until it'll become mainstream.
  https://patchwork.ozlabs.org/project/buildroot/list/?series=433071

**Comment by ogalbxela** [comment]
Thank you for your contribution. Below is some background information to clarify the concerns you're addressing with your patches.

Mstflint is an open-source part of NVIDIA firmware tools (https://network.nvidia.com/products/adapter-software/firmware-tools/ - a proprietary code). Our typical process involves incorporating certain internal product features into the mstflint codebase with each release. While mstflint releases generally (but not always) align with mft releases, the mft undergoes a thorough testing cycle across various os-distribution-adapter combinations. For mstflint we only perform basic functionality testing.

We do not deploy dynamic libraries for either mft or mstflint as well as we do not test neither of products in that configuration, all tools are statically linked with all its components. If you are aiming to link your code with the mtcr_ul library, please use its static version. Unfortunately, we cannot release something that has not been tested. Changing our internal process with this regard may take a little bit less than forever.

Regarding your concern about the path used for placing the library: both mft and mstflint deploy its own versions of mtrc_ul library. To distinguish between them when both products are installed, we place the open-source version under /usr/lib/mstflint.

To summarize: we can only consider the portion of your patch related to the hardcoded paths. Additionally, we do not work directly on the master branch. All development is done on the master_devel. For each patch from the community, we first absorb it to the internal product, subject it to our internal test cycle, and then it becomes a part of official release. This usually happens (though not always) once every three months.

**Comment by vjardin** [comment]
Thanks @ogalbxela  for the feedbacks.

The goal is to properly support DPDK with Buildroot, see this series:
  https://patchwork.ozlabs.org/project/buildroot/list/?series=433904

so, once mstflint, libmtcr_ul.so is available, DPDK's meson can probe it thanks to:
  https://patchwork.ozlabs.org/project/buildroot/patch/20241122230120.96898-3-vjardin@free.fr/
needed by:
  https://git.dpdk.org/dpdk/tree/drivers/common/mlx5/linux/meson.build#n48

Do you want me to rebase this patch on the branch `master_devel` ?

Regarding dynamic vs static: what's about having a Makefile variable ? By default, it would be static, so it does not break your release and test processes ; but then, on special cases we can enable a build without static ?

**Comment by vjardin** [comment]
Please @ogalbxela , do you have some feedbacks ?

**Comment by ogalbxela** [comment]
I'll bring this up with our management team and let you know.

**Comment by vjardin** [comment]
> what is preventing you from statically linking mtcr_ul? 

issues with the current static linking:
  - it did not work on Buildroot with DPDK that is looking for the `.so`; so I'd prefer to fix mstflint configure/makefiles which have this issue any many other ones
  - it adds a memory footprint overhead because I do run the system on Ramdisk in order to avoid any issues with the physical disks

And frankly, why would you support static linking only nowadays ?

We need the solution to be more flexible to be compiled into some embedded environments. For my case, it is Buildroot, but I guess that some others would have some similar issues for some other embedded Linux.

**Comment by tzafrir-mellanox** [comment]
Any news on this one?

Some of the fixes here seem useful regardless of static vs. dynamic: patches 2, 3 and 7 (libxml location, libcurl location and the extra missing library). Are those indeed useful as-is and can be merged separately?
